test/Exporter/Test/Writer/KnpSnappyWriterTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+// Use same namespace as system under test to mock native functions
+
+namespace Exporter\Writer;
+
+use org\bovigo\vfs\vfsStream;
+use org\bovigo\vfs\vfsStreamDirectory;
+use org\bovigo\vfs\vfsStreamWrapper;
+use Prophecy\Argument;
+
+function is_executable()
```

Comment:
@jordisala1991 He is using a [virtual filesystem] to test the Writer class properly (without testing any dependencies). Integration tests (that test the whole behavior) are located in the Integration sub-namespace. Unfortunately (as you may know) PDF export is very hard to make "generic" and so it could lead to unexpected results in an admin context, this will not be merged.